### PR TITLE
MySQL second pass cleanup

### PR DESCRIPTION
**Pull request by mbenmclean**
This is a second pass on https://github.com/verrazzano/verrazzano/commit/34bf6e23945a2cd00aa275c3de462206f073aac3
and resolves comments from https://github.com/verrazzano/verrazzano/pull/2045
Contents:
- Leverages the operation field in the component context
- Moves findVolumeTemplate to common code and removes all duplicate non-exported functions
- Cleans up misplaced constant
- Cleans up typo

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/mysql/mysql.go [hunk 1]
```diff
@@ -60,17 +56,17 @@ func isEnabled(context spi.ComponentContext) bool {
...
-		Namespace: vzconst.KeycloakNamespace,
-		Name:      secretName}
-	// Get the mysql secret
-	err := compContext.Client().Get(context.TODO(), nsName, secret)
-	if err != nil && !k8serror.IsNotFound(err) {
-		return []bom.KeyValue{}, ctrlerrors.RetryableError{Source: ComponentName, Cause: err}
-	}
-	// see if the mysql exists
-	if !k8serror.IsNotFound(err) {
+	if compContext.For(ComponentName).GetOperation() == "upgrade" {
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/component/mysql/mysql.go:59]: We should put constants for these in the platform-operator/constants/constants.go

File: platform-operator/controllers/verrazzano/component/mysql/mysql.go [hunk 2]
```diff
@@ -60,17 +56,17 @@ func isEnabled(context spi.ComponentContext) bool {
...
+		secret := &v1.Secret{}
+		nsName := types.NamespacedName{
+			Namespace: vzconst.KeycloakNamespace,
+			Name:      secretName}
+		// Get the mysql secret
+		err := compContext.Client().Get(context.TODO(), nsName, secret)
+		if err != nil {
+			compContext.Log().Errorf("Error getting mysql secret: %v", err)
+			return []bom.KeyValue{}, ctrlerrors.RetryableError{Source: ComponentName, Cause: err}
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/component/mysql/mysql.go:68]: This is the right thing to do, but for now will still cause us to fail upgrade until we fix that.

File: platform-operator/controllers/verrazzano/component/mysql/mysql.go [hunk 3]
```diff
@@ -84,12 +80,7 @@ func appendMySQLOverrides(compContext spi.ComponentContext, _ string, _ string,
 
 	kvs = append(kvs, bom.KeyValue{Key: mySQLUsernameKey, Value: mySQLUsername})
 
-	// See if MySQL is deployed and create the MySQL init file if not
-	deployed, err := helmutil.IsReleaseDeployed(ComponentName, vzconst.KeycloakNamespace)
-	if err != nil {
-		return []bom.KeyValue{}, ctrlerrors.RetryableError{Source: ComponentName, Cause: err}
-	}
-	if !deployed {
+	if compContext.For(ComponentName).GetOperation() == "install" {
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/component/mysql/mysql.go:83]: Same comment, we should create constants for the op names

**Review by michael-cico — APPROVED** [review]

File: platform-operator/controllers/verrazzano/component/mysql/mysql.go (hunk 2 above)
Comment (mbenmclean) [reply platform-operator/controllers/verrazzano/component/mysql/mysql.go:68]: Well what the desired behavior. We could either fail upgrade or log that the mysql secret was not found so the password will be reset. What do you think

Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/mysql/mysql.go:70]: My comment was I htink this is the right thing to do, lol.  I probably should have stayed silent :-)

**Pull request merged**